### PR TITLE
Setup up ember-data

### PR DESCRIPTION
**Pull request by bcardarella**
Assume ember-data by default.
Assume ActiveModelAdapter
Encourage best-practice of versioning API

### CODE REVIEW

**Comment by bcardarella** [comment]
I'm willing to let this one sit here for a bit before I merge in so we can discuss.

**Comment by bcardarella** [comment]
It would be really nice if we could override the regular Rails generators to only produce the json response, not generate any views, and namespace under `api/vX`

**Comment by bcardarella** [comment]
Satisfies #30

**Comment by bcardarella** [comment]
I had a thought last night. Perhaps there could be an app config setting. Something like:

``` ruby
config.ember.data.api_version = 2
```

Then the api version in `store.js.erb` would be

``` erb
namespace: 'api/<%= api_version %>'
```

In addition, any Rails resources/controllers would be auto generated under this API version number. You could opt-out of the namespacing with a `--not-api` switch.

**Comment by twokul** [comment]
@bcardarella I'm going to second `app.config`.

**Comment by bcardarella** [comment]
I'm re-thinking the api_version commit. I forgot how much of a problem this will be with the asset pipeline. If you change the version number the store.js asset will not recompile because that asset has not actually changed. Unless there is a way to get the state of that file to also subscribe to the api_version number as a signal that it has change I don't see this as a viable option. We can default to `api/v1`

File: test/generators/bootstrap_generator_test.rb [hunk 1]
```diff
@@ -48,6 +52,17 @@ class BootstrapGeneratorTest < Rails::Generators::TestCase
       run_generator
       assert_file "#{custom_path}/ember-app.js"
       assert_file "#{custom_path}/router.js.es6"
+      assert_file "#{custom_path}/store.js.erb"
     end
   end
+
+  test "Uses config.ember.ember.data.api_version" do
+    original_version = Rails.application.config.ember_appkit.data.api_version
```
Comment (rwjblue) [inline test/generators/bootstrap_generator_test.rb]: We have a nice helper for ensuring that these are unset.  It is `with_config` and defined [here](https://github.com/dockyard/ember-appkit-rails/blob/master/test/support/generator_test_support.rb#L15-L28).

Example is [here](https://github.com/dockyard/ember-appkit-rails/blob/master/test/generators/bootstrap_generator_test.rb#L47-L51).

**Comment by bcardarella** [comment]
I'm merging!

**Pull request merged**